### PR TITLE
Adding missing annotations and conditional check got deferred

### PR DESCRIPTION
**Pull request by jacobjzhang**
Associated Issue: #1034
### Summary of Changes
- Adding missing flow type annotations and conditional check got deferred
### Testing
- [X] passes `npm test`
- [X] passes `npm run lint`

### CODE REVIEW

File: packages/devtools-config/src/feature.js [hunk 1]
```diff
@@ -7,11 +7,11 @@ const flag = require("./test-flag");
  * Gets a config value for a given key
  * e.g "chrome.webSocketPort"
  */
-function getValue(key) {
+function getValue(key :mixed) {
   return pick(config, key);
 }
 
-function isEnabled(key) {
+function isEnabled(key :mixed) {
```
Comment (jasonLaster) [inline packages/devtools-config/src/feature.js]: This should be a strIng 

Nit: param colon space type

File: public/js/actions/sources.js [hunk 2]
```diff
@@ -317,8 +317,11 @@ function getTextForSources(actors: any[]) {
     function maybeFinish() {
       if (pending.size == 0) {
         // Sort the fetched sources alphabetically by their url.
-        deferred.resolve(
-          fetched.sort(([aFirst], [aSecond]) => aFirst > aSecond ? -1 : 1));
+        if (deferred) {
+          deferred.resolve(
+            fetched.sort(([aFirst], [aSecond]) => aFirst > aSecond ? -1 : 1)
+          );
+        }
```
Comment (jasonLaster) [inline public/js/actions/sources.js]: Why do we need the if?

File: public/js/utils/assert.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function assert(condition, message) {
+function assert(condition :any, message: string) {
```
Comment (jasonLaster) [inline public/js/utils/assert.js]: I think the condition is a boolean

File: public/js/actions/sources.js (hunk 2 above)
Comment (jacobjzhang) [inline public/js/actions/sources.js]: deferred is assigned defer() at the beginning of the script which has the possibility of being undefined, one of the things flowtype pointed out actually.

**Pull request merged**